redash/query_runner/azure_kusto.py

```diff
@@ -0,0 +1,165 @@
+                },
+                "database": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "cluster", "azure_ad_client_id", "azure_ad_client_secret",
+                "azure_ad_tenant_id", "database"
+            ],
+            "order": [
+                "cluster", "azure_ad_client_id", "azure_ad_client_secret",
+                "azure_ad_tenant_id", "database"
+            ],
+            "secret": ["azure_ad_client_secret"]
+        }
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    @classmethod
+    def type(cls):
+        return "azure_kusto"
+
+    @classmethod
+    def name(cls):
+        return "Azure Data Explorer (Kusto)"
+
+    def run_query(self, query, user):
+
+        kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(
+            connection_string=self.configuration['cluster'],
+            aad_app_id=self.configuration['azure_ad_client_id'],
+            app_key=self.configuration['azure_ad_client_secret'],
+            authority_id=self.configuration['azure_ad_tenant_id'])
+
+        client = KustoClient(kcsb)
+
+        db = self.configuration['database']
+        try:
+            response = client.execute(db, query)
+
+            result_cols = response.primary_results[0].columns
+            result_rows = response.primary_results[0].rows
+
+            columns = []
+            rows = []
+            for c in result_cols:
+                columns.append({
+                    'name': c.column_name,
+                    'friendly_name': c.column_name,
+                    'type': TYPES_MAP.get(c.column_type, None)
+                })
+
+            # rows must be [{'column1': value, 'column2': value}]
+            for row in result_rows:
+                rows.append(row.to_dict())
+
+            error = None
+            data = {'columns': columns, 'rows': rows}
+            json_data = json_dumps(data)
+
+        except KustoServiceError as err:
+            json_data = None
+            try:
+                error = err.args[1][0]['error']['@message']
+            except (IndexError, KeyError):
+                error = err.args[1]
+        except KeyboardInterrupt:
+            json_data = None
+            error = "Query cancelled by user."
+
+        return json_data, error
+
+    def get_schema(self, get_stats=False):
+
+        get_tables_list_query = ".show tables | project TableName"
+
+        get_tables_list_results, error = self.run_query(get_tables_list_query, None)
+        if error is not None:
+            raise Exception("Failed getting schema.")
+        tables_list_response = json_loads(get_tables_list_results)
+
+        tables_list = []
+
+        for table in tables_list_response['rows']:
+            get_table_schema_query = ".show table %s schema as json | project Schema" % table['TableName']
```

Comment:
@danield137 can I somehow get the full json schema in one query? 
if only `.show table`**s**` schema as json` would work instead of `show table <table> schema`

Reply:
`.show database schema as json`

Reply:
😭Ok I will fix this now.